src/lib.rs

```diff
@@ -8,97 +8,118 @@ pub struct DB<T> {
 /// NB: (nota bene, or "take special note"): You will need to be explcit about the liftimes in this
 /// struct
 #[derive(Debug, PartialEq, Eq)]
-pub struct DBView<T> {
-    entries: Vec<&T>,
+pub struct DBView<'a, T:'a> {
+    entries: Vec<&'a T>
 }
 
 /// An mutably borrowed subset of a DB
 ///
 /// NB: You will need to be explcit about the liftimes in this struct
 #[derive(Debug, PartialEq, Eq)]
-pub struct DBViewMut<T> {
-    entries: Vec<&mut T>,
+pub struct DBViewMut<'a, T> where T : 'a {
+    entries: Vec<&'a mut T>,
 }
 
 /// Filters a DBView using the the given predicate.
-pub fn filter_one<T, F>(view: &DBView<T>, predicate: F) -> DBView< T>
+pub fn filter_one<'a, 'b, T, F>(view: &'b DBView<'a, T>, predicate: F) -> DBView<'a, T>
     where F: Fn(&T) -> bool
 {
-    unimplemented!()
+    view.select_where(predicate)
 }
 
 /// Filters two DBView structs using the same predicate, producing two separate results. This is
 /// the moral equivalent of doing the two filters separately.
-pub fn filter_two<T, F>(view_a: &DBView<T>,
-                        view_b: &DBView<T>,
+pub fn filter_two<'a, 'b, 'c, T, F>(view_a: &'c DBView<'a, T>,
+                        view_b: &'c DBView<'b, T>,
```

Comment:
So saying `'c` should be the lifetime of both references *seems* like an unnecessary constraint -- there's no reason they shouldn't be independent.

It turns out that using `'c` for both still allows the actual parameters to be independent in this case. Think about this! (or ask me).